azure_pipeline.yaml

```diff
@@ -13,7 +13,58 @@ variables:
   action:
 
 stages:
+  - stage: Precheck
+    jobs:
+      - job: Validate
+        pool:
+          vmImage: $(agentPool)
+        steps:
+        - task: AzureKeyVault@1
+          inputs:
+            ConnectedServiceName: azurerm-sandbox
+            keyVaultName: infra-vault-nonprod
+            secretsFilter: 'azure-devops-token'
+            runAsPreJob: false
+        - task: Bash@3
+          env:
+            thisbuild: $(Build.BuildId)
+          inputs:
+            targetType: 'inline'
+            script: | 
+              #!/bin/bash
+              set -e
+
+              organization="hmcts"
+              project="CNP"
+
+              echo "This build number is: $thisbuild"
+
+              while [[ $thisbuild -ne $buildrunning ]]
+              do
+              JSON_DATA=$(curl -u :$(azure-devops-token) --request GET "https://dev.azure.com/$organization/$project/_apis/build/builds?api-version=5.1&definitions=235" -H "Content-Type: application/json" | jq  '.value[] | .status + (.id|tostring)' | sort -u)
+
+              for value in ${JSON_DATA[@]}; 
+              do 
+
+              number=(${value//[!0-9]/})
+              text=(${value//[0-9]/})
+              textcheck=$(sed -e 's/^"//' -e 's/"$//' <<<"$text")
+
+              if [[ "$textcheck" == "inProgress" ]];  then
+              sleep 2m
```

Comment:
is it an expensive api call? 2min is quite a long wait

Reply:
No, added 2m so it doesn't hit the api every few seconds. Added just to delay the task

Reply:
I think the issue is that frontdoor pipeline will run for 1 hour before it fails. So we want to limit the api calls as we definitely know that frontdoor will be in progress for 1 hour. we probably will revisit it once we have separated the pipeline.

Reply:
it's not just frontdoor it could be anything, i mean why not check a bit more often? every 30s or every 10s ?

Reply:
@timja changed to 30s as discussed